calculator.rb

```diff
@@ -0,0 +1,101 @@
+puts "HELLO I AM CALCU-BOT. \nSELECT A MATH OPERATION."
+
+operands = [
+{id: 1, name: "ADDITION", symbol: "+"},
+{id: 2, name: "SUBTRACTION", symbol: "-"},
+{id: 3, name: "MULTIPLICATION", symbol: "*"},
+{id: 4, name: "DIVISION", symbol: "/"},
+{id: 5, name: "EXPONENTIATION", symbol: "^"},
+{id: 6, name: "MODULO", symbol: "%"}
+]
+
+operands.each do |operand|
+  puts "#{operand[:id]}. #{operand[:name]} #{operand[:symbol]}"
+end
+
+# will be assigned user-inputted operator
+math_operand = "none"
+
+while math_operand == "none"
+  user_operand = gets.chomp.upcase
+  
+  # truncate user_operand for use in case block
+  # accounts for if a user inputs "divide" or "division" (both begin with DIV)
+  user_operand = user_operand[0..2]
+  
+  case user_operand
+  when "1", "ADD", "+"
+    puts "YOU HAVE CHOSEN ADDITION."
+    math_operand = :+
+  when "2", "SUB", "-"
+    puts "YOU HAVE CHOSEN SUBTRACTION."
+    math_operand = :-
+  when "3", "MUL", "*"
+    puts "YOU HAVE CHOSEN MULTIPLICATION."
+    math_operand = :*
+  when "4", "DIV", "/"
+    puts "YOU HAVE CHOSEN DIVISION."
+    math_operand = :/
+  when "5", "EXP", "^"
+    puts "YOU HAVE CHOSEN EXPONENTIATION."
+    math_operand = :**
+  when "6", "MOD", "%"
+    puts "YOU HAVE CHOSEN MODULO-IFICATION. \n(CALCU-BOT DOES NOT KNOW THE VERB TENSE FOR A MODULO.)"
+    math_operand = :%
+  else
+    puts "CALCU-BOT HAS ENCOUNTERED AN ERROR. SELECT A MATH OPERATION."
+    math_operand = "none"
+  end
+end
+
+# ensure a value is an integer or float
+class String
+  def integer_checker
+    self.to_i.to_s == self
+  end
+  def float_checker
+    self.to_f.to_s == self
+  end
+end
+
+
+# returns error message until user inputs an integer or float
+def class_check(num)
```

Comment:
Minor suggestion: I tend to really like having my method names read more like verbs. What is this method responsible for? It's used for checking what kind of number it is. There are a lot of different good names for this, but maybe I would name this `check_valid_number_type` or something like that